### PR TITLE
backport: Fix missing error class and test for out_forward

### PR DESCRIPTION
**Pull request by okkez**
This backports #922

### CODE REVIEW

File: lib/fluent/output.rb [hunk 1]
```diff
@@ -450,9 +450,17 @@ def initialize
       super
     end
 
+    def format_stream(tag, es)
+      if @time_as_integer
```
Comment (repeatedly) [inline lib/fluent/output.rb]: `time_as_integer` doesn't have this option so this commit should not be backported to v0.12

Comment (okkez) [inline lib/fluent/output.rb]: I've removed `@time_as_integer`.

**Comment by repeatedly** [comment]
@tagomoris Can we merge this?

File: lib/fluent/output.rb [hunk 2]
```diff
@@ -450,9 +450,13 @@ def initialize
...
     end
 
+    def format_stream(tag, es)
+      es.to_msgpack_stream
+    end
+
     def emit(tag, es, chain)
       @emit_count += 1
-      data = es.to_msgpack_stream
+      data = format_stream(tag, es)
```
Comment (repeatedly) [inline lib/fluent/output.rb]: Extracting routine to `format_stream` is for `@time_as_integer` so this diff itself is not needed for this patch?
We want to reduce the change without no reason.

**Comment by tagomoris** [comment]
Honestly speaking, I'm not sure this patch is needed for v0.12 branch. This change is important to add further fixes on these code/plugins, but `v0.12` branch is not development branch.

Code and merging this itself is ok for me.

**Comment by repeatedly** [comment]
`ForwardOutputConnectionClosedError` definition is needed for v0.12, right?

**Comment by tagomoris** [comment]
@repeatedly Right. But it's just a bug, and its fix should be separated from test-driver fixes ideally, especially in stable branch.

**Comment by repeatedly** [comment]
@tagomoris okay.

@okkez Could you create a new PR?

**Comment by okkez** [comment]
@repeatedly OK. Do we need 2 new PRs?

* Just add `ForwardOutputConnectionClosedError`
* Except above change in this PR

**Pull request closed without merging**